src/main/kotlin/org/rust/ide/utils/SearchByOffset.kt

```diff
@@ -55,6 +55,20 @@ fun findExpressionInRange(file: PsiFile, startOffset: Int, endOffset: Int): RsEx
  * as return expression, doesn't allow attrs and items) within selected range.
  */
 fun findStatementsInRange(file: PsiFile, startOffset: Int, endOffset: Int): Array<out PsiElement> {
+    val elements = findElementsInRange(file, startOffset, endOffset)
+
+    // Finally check if found elements meet requirements, and return result
+    elements.forEachIndexed { idx, element ->
```

Comment:
I think this loop can be replaced by two checks: 

1) drop last element and use `.any` to check that they are statements or comments.
2) look at the last element specifically.